store/src/sql/event.rs

```diff
@@ -552,10 +552,11 @@ impl ceramic_api::AccessModelStore for SqliteEventStore {
         end: &EventId,
         offset: usize,
         limit: usize,
-    ) -> anyhow::Result<Vec<(EventId, Vec<u8>)>> {
+    ) -> anyhow::Result<Vec<(Cid, Vec<u8>)>> {
         let res = self
             .range_with_values_int(start, end, offset, limit)
-            .await?;
+            .await?
+            .map(|(event_id, value)| (event_id.cid().unwrap(), value));
```

Comment:
how bad do yall think this is?  It's a bit inefficient.  Another approach would be to update the SQL query to return the CID and then thread it through here so we get the CID from the SQL result instead of by parsing the eventID.  The problem with that approach is that the `range_with_values` method exists for both the `AccessModelStore` and `Recon::Store` traits, and the later needs it to continue to return EventIds not Cids, so if I want to take that approach I will need to add a new method with a different name that is only used by the AccessModelStore and returns Cids.  I can definitely make that change though if you think it's the better way to go.  @dav1do @nathanielc

Reply:
I'm not worried about the performance of this right now, but I wonder if we can/should change EventId to be more complex than a `Vec<u8>` wrapper? It requires a CID to build, so returning None is pretty annoying to have all these unwraps.. and if it can't be None, it seems like it shouldn't return an option. Seeing the `unwrap` is definitely a red flag.. and usually something you want to comment like you would unsafe. Crashing the entire process over bad data would be _bad_, even if that data _should_ be impossible. 

My instinct here is to return remove them (or just even just error) even though they should never exist, but I'd rather EventId be more complex and actually store the its parts and then have a `to_vec()` method. Curious to know the reason it's like this before changing it though.

```rust
        let res = self
            .range_with_values_int(start, end, offset, limit)
            .await?
            .filter_map(|(event_id, value)| {
                if let Some(cid) = event_id.cid() {
                    Some((cid, value))
                } else {
                    warn!(?event_id, "Invalid event ID missing CID");
                    None
                }
            });
```

Your point about whether we should thread CID through is interesting... I don't think you need to do it now, but it'd be nice to draw the boundary between recon and the event store more clearly so we aren't parsing the CID in/out of the EventId on both sides. 

I'm looking at that as part of IOD and consuming these changes. My current expectation is that the store will no longer implement any of the existing traits directly, and the ceramic service (or whatever it's called) will be the one that turns bytes into events and then calls into the store, so the store could take a ValidatedEvent or whatever makes the input/output easier to work with. 

My general philosophy is simple, correct, performant. Rust makes it more obvious than most languages that you're not doing something "the most optimal" way, but it's hard to know how bad it is without profiling. Rust and LLVM are pretty good about removing clones and using cache, so outside of the bad algorithm/obvious stuff, I try to avoid optimizing in favor of simplicity/clarity until it's proven to be an issue.

Reply:
Lets not change the EventId yet, its a thin wrapper around a Vec<u8> so it can lexicographically sort. The option returned is that it is technically possible to construct an EventId from bytes that do not decode correctly into the CID. So while annoying I say we explicitly handle the case where there is no CID in the API code. 

Something like:

```
.map(|(event_id, value)| Ok((event_id.cid().ok_or_else(|| anyhow!("EventId does not have an event CID")?, value)))?;
```


Since we are gradually reducing the surface area of the EventId type itself, we can likely change it or remove it entirely from the API soon. Given that I say we wait on any more refactors of its type.